### DISCUSSION TITLE
Restricting implied relationships to same hierarchy

### DISCUSSION THREAD

**Discussion by hornjbl** [SkepticMystic/breadcrumbs#432] [Q&A]
Is there a way to tell the plugin to pick up implied relationships ONLY within the same hierarchy? 

What I'm seeing now is the following: 

I have a hierarchy of, for example, parent-child, and another of employer-employee.

If A is the mother of B, but also the employer of X, then the plugin seems to think that B is the implied sibling of X, because they are both "under" A. Even though the relationships aren't in the same hierarchy, and therefore shouldn't be forming the implied relationships between B and X.

I still want the implied relationships within the same hierarchy though. So if I specified A as the mother of both B and C, the implied relationship of sibling between B and C should still be formed.

Looking through the settings, it looks like you can turn implied relationships off and on on various levels, but not restrict it to within the same hierarchy.

**Comment by michaelpporter** [comment]
This is handled in v4 by the way transitive rules work: **each rule matches specific named fields, so implied relations don't cross between hierarchies** as long as you give each hierarchy its own fields.

In your example, use distinct fields per hierarchy:

```
# family
mother / child  (with a sibling rule scoped to them, e.g. [child, mother] -> sibling)

# work
employer / employee
```

A rule like `[child, mother] -> sibling` only ever traverses `child` then `mother` edges, so B (child of A) and X (employee of A) will **not** become implied siblings — the work edges aren't part of that chain. The v3 bleed you're describing came from every hierarchy sharing the generic `up`/`same`/`down` fields; in v4 you avoid it by separating fields and only adding the cross-field rules you actually want.

Closing as resolved — shout if a specific rule still bleeds across hierarchies.
